### PR TITLE
Set file permissions

### PR DESCRIPTION
**Pull request by aritosteles**
|Related issue|
|---|
|#20727|

## Description
Modified InstallerScripts.vbs and wazuh-installer.wxs. 

Added a custom function that sets file permissions.
Added function SetPermissions() to InstallerScripts.vbs. This function enables inheritance for the installation folder and sets Read and Execute permission for the folder and all its files and subfolders.

### Evidence:

Installation of v4.7.4-1 and subsequent upgrade to 4.9.1:
[Screencast from 2024-05-09 18-55-52.webm](https://github.com/wazuh/wazuh/assets/100874572/3170d4d3-2ef5-4122-97cf-3785ba1c7955)
 
Upgrade to 4.9.2 following previous test:
[Screencast from 2024-05-09 22-18-23.webm](https://github.com/wazuh/wazuh/assets/100874572/2b391b16-b52f-4ff8-99f3-8e836f59cf53)

Uninstall of previously installed version. Clean 4.9.1 installation. Upgrade to 4.9.2:
[Screencast from 2024-05-09 22-23-21.webm](https://github.com/wazuh/wazuh/assets/100874572/267e7361-ad80-43ab-8747-277e903a1600)

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [x] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [x] Package installation
- [ ] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

**Review by tdrauncieal — CHANGES_REQUESTED** [review]

File: src/win32/wazuh-installer.wxs [hunk 1]
```diff
@@ -115,6 +115,9 @@
         <CustomAction Id="SetCustomActionDataValue" Return="check" Property="CustomAction_InstallerScripts" Value="&quot;[APPLICATIONFOLDER]&quot;/+/&quot;[OS_VERSION]&quot;/+/&quot;[WAZUH_MANAGER]&quot;/+/&quot;[WAZUH_MANAGER_PORT]&quot;/+/&quot;[WAZUH_PROTOCOL]&quot;/+/&quot;[NOTIFY_TIME]&quot;/+/&quot;[WAZUH_REGISTRATION_SERVER]&quot;/+/&quot;[WAZUH_REGISTRATION_PORT]&quot;/+/&quot;[WAZUH_REGISTRATION_PASSWORD]&quot;/+/&quot;[WAZUH_KEEP_ALIVE_INTERVAL]&quot;/+/&quot;[WAZUH_TIME_RECONNECT]&quot;/+/&quot;[WAZUH_REGISTRATION_CA]&quot;/+/&quot;[WAZUH_REGISTRATION_CERTIFICATE]&quot;/+/&quot;[WAZUH_REGISTRATION_KEY]&quot;/+/&quot;[WAZUH_AGENT_NAME]&quot;/+/&quot;[WAZUH_AGENT_GROUP]&quot;/+/&quot;[ENROLLMENT_DELAY]&quot;" />
         <CustomAction Id="CustomAction_InstallerScripts" BinaryKey="InstallerScripts" VBScriptCall="config" Return="check" Execute="commit" Impersonate="no"/>
 
+        <CustomAction Id="SetCustomActionDataValue2" Return="check" Property="SetPermissions" Value="&quot;[APPLICATIONFOLDER]&quot;" />
```
Comment (ncvicchi) [inline src/win32/wazuh-installer.wxs:118]: Reusing names as "SetCustomActionDataValue" to "SetCustomActionDataValue2" tends to ofuscate code in time and might grown badly. If this value is used exclusively with the setPermissions functionality I would use some name that would refer to it as "SetPermissionsCustomActionDataValue" or similar to enforce relationship

File: src/win32/wazuh-installer.wxs [hunk 2]
```diff
@@ -179,8 +182,13 @@
             <!-- Stop the service as soon as possible on uninstall, and only on uninstall, in order to unlock the files and folders that must be deleted. -->
 			<Custom Action="SetRemoveAllDataValue" Before="InstallFinalize">(NOT UPGRADINGPRODUCTCODE) AND (REMOVE="ALL")</Custom>
  		    <Custom Action="CustomAction_RemoveAllScript" After="SetRemoveAllDataValue">(NOT UPGRADINGPRODUCTCODE) AND (REMOVE="ALL")</Custom>
+ 
+            <Custom Action="SetCustomActionDataValue2" Before="InstallFinalize"/>
```
Comment (Leoquicenoz) [inline src/win32/wazuh-installer.wxs:186]: You could add the closing of the `<Custom Action></Custom>`, this would keep the same existing syntax.

**Review by Leoquicenoz — CHANGES_REQUESTED** [review]

**Review by tdrauncieal — APPROVED** [review]
LGTM.

**Review by ncvicchi — APPROVED** [review]
LGTM!

**Review by Leoquicenoz — APPROVED** [review]
LGTM!!

**Comment by Dwordcito** [comment]
The failed checks are pre-existing and are due to the fact that 4.9.0 must be updated with the changes introduced in 4.8.0

**Pull request merged**